### PR TITLE
Custom web component "<adr-report></adr-report>" methods to fetch ADR report

### PR DESCRIPTION
**Pull request by phchen95**
PR addresses:

1. Include a method to get the report `guid`
2. Include methods to define and call a custom web component `<adr-report>` to fetch ADR report
3. Add test cases for the added methods.

### CODE REVIEW

File: src/ansys/dynamicreporting/core/adr_report.py [hunk 1]
```diff
@@ -171,6 +171,332 @@ def get_url(self, filter: str = "") -> str:
...
+            report_url = my_report.get_guid()
+        """
+        guid = ""
+        if self.service is None:  # pragma: no cover
+            self.service.logger.error("No connection to any report")
+            return guid
+        if self.service.serverobj is None:  # pragma: no cover
+            self.service.logger.error("No connection to any server")
+            return guid
+        if self.service.url is None:
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/adr_report.py:200]: Suggestion: Combine this if block and the ones before using 'or'

File: src/ansys/dynamicreporting/core/adr_report.py [hunk 2]
```diff
@@ -171,6 +171,332 @@ def get_url(self, filter: str = "") -> str:
...
+            import ansys.dynamicreporting.core as adr
+            adr_service = adr.Service(ansys_installation = r'C:\\Program Files\\ANSYS Inc\\v232')
+            ret = adr_service.connect()
+            my_report = adr_service.get_report(report_name = 'Top report')
+            my_report.get_report_component()
+        """
+        # fetch method using predefined prefix rules in the proxy server OR using traditional <iframe>
+        # add host-style-path attribute if specified (can only work when prefix is provided)
+        host_style_path = f'host-style-path="{style_path}"' if style_path else ""
+        fetchMethod = (
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/adr_report.py:492]: snake_case here

**Review by viseshrp — COMMENTED** [review]
Some minor things, mostly LGTM

File: src/ansys/dynamicreporting/core/adr_report.py (hunk 2 above)
Comment (phchen95) [reply src/ansys/dynamicreporting/core/adr_report.py]: Thanks for spotting this, writing JS for so long have got me default to camelCase

Comment (viseshrp) [reply src/ansys/dynamicreporting/core/adr_report.py]: @phchen95 lol no worries, I can totally relate to this

**Review by viseshrp — APPROVED** [review]

**Pull request merged**